Feature/underground 2.1

AevitasDragonkin opened the pull request:
## Description
### UI
- [X] Updated Dig UI
- [X] Updated Treasures UI
- [X] Minimum and maximum items per layer is now displayed in a table format in the Help tab
- [X] Added images to Underground Helpers
- [X] Removed decimal point on UG and helper XP display
- [X] Centered Expert data on their info tab

### Textual
- [X] Refactored discovery time explanation in the Help tab
- [X] Refactored Special mine trigger requirements in the Help tab
- [X] Refactored the Energy Restore description to mention Helpers explicitly
- [X] Refactored Veteran Aristo's flavor text to the correct diamond cost for Ice Memory
- [X] Explained what uses Underground Level brings in the EXP section
- [X] Fixed Medichamite helper text to mention it starting at level 16 instead of 15
- [X] Renamed all mines to give them more flavor in their naming
- [X] Renamed "Item" to "Treasure"
- [X] Renamed "Found" treasure to "Collected"
- [X] Renamed "Lexicon" to "Glossary"
- [X] Renamed "Underground Helper" to "Underground Miner"
- [X] Renamed "Favorite Mine" to "Expertise"
- [X] Renamed "Look for Favorite mine" to "Use Expertise"
- [X] Added custom steal messages for each Underground Helper
- [X] Fix quests mentioning layers instead of full mines

### Behaviour
- [X] Tools now restore durability with each game tick
- [X] Tools now have a chance to destroy the item and award no XP when it digs up the item (Bomb 100%, other tools 0%)
- [X] Survey now prioritises unseen treasure
- [X] Survey area highlighting is now removed when the item is collected
- [X] Survey restore rate starts increasing from level 15
- [X] Refactored and reset Underground Daily Trades to Underground Trades from the statistics and statistics display
- [X] Underground Helpers can now select multiple energy restores to use
- [X] Added Battery Discharge keybind
- [X] Now notifying the player when successfully selling, or trading, items in the Underground
- [X] Underground Treasures tab now has a quick sell all mode
- [X] Palaeo token can now be sold for diamonds
- [X] Different Discharges are now tracked via statistics
- [X] Different types of mine fully cleared are now tracked via statistics
- [X] Different items found are now tracked via statistics
- [X] Number of items stolen by each helper is now tracked
- [X] Battery is now available for use without equipping the Cell Battery Oak Item

### Bug Fixes
- [X] Fixed Generic Trades button disabled when going from 11 or 101 to 1 available trade
- [X] Fixed discharge button being available when on discovery time
- [X] Palaeo token no longer shows up in Berry deals

### Balancing
- [X] Restore prices scaled: Small (100%), Medium (75%), Large (50%)
- [X] The tool restore rate formula has been made slightly smoother for the higher levels
- [X] Cell Battery Oak Item will reduce the charges needed to discharge
- [X] Cell Battery level up requirements have been scaled up slightly to account for the max charge reduction
- [X] Discharge Patterns now have a UG level requirement instead of a Cell Battery requirement

Not included in this PR and held for another future PR:
- Text visibility on various themes
- Underground Helper content past level 50
- Survey and Chisel becoming redundant past a certain level


## How Has This Been Tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->
Tested each change on my own, needs testing from the beta group


## Closes
Closes #5777 #5713 #5733 #5702 #5692 #5681 


## Types of changes
<!-- What types of changes does your code introduce? Delete any that don't apply, and/or add more you think would be useful -->
- Bug fix
- Feedback implementation

CypherX commented:
Not sure if it's just a dev build thing but the [Show Locked Items] toggle is also showing items from Hisui and Paldea, can we confirm and if needed limit it to items from released regions?
![image](https://github.com/user-attachments/assets/91e763f3-7c9e-42de-a78b-cdbe7e938ab5)


Also, on certain themes the toggle switch is appearing on the next line due to text size, maybe add the `small` class to the [Show Locked Items] text:
![image](https://github.com/user-attachments/assets/1a06ca18-fa85-4339-80e2-13ae764df0fd)

CypherX commented:
The changed tool borders and "infinite" animation don't appear to work in the client:

![image](https://github.com/user-attachments/assets/ec8a1ce0-2126-4b9c-8b7d-12ba8f4e5cf4)
![image](https://github.com/user-attachments/assets/2ff2a439-2488-4ea9-9675-b05264014ef3)

AevitasDragonkin commented:
> The changed tool borders and "infinite" animation don't appear to work in the client:

Fixed

> Not sure if it's just a dev build thing but the [Show Locked Items] toggle is also showing items from Hisui and Paldea, can we confirm and if needed limit it to items from released regions? 

We'll fix by updating the MaxRegionRequirement and StoneUnlockRequirement to show a better hint. The item can stay visible as it's already showing up in Berry Trades and in the Evolution Item tab.

> Also, on certain themes the toggle switch is appearing on the next line due to text size, maybe add the `small` class to the [Show Locked Items] text

Fixed
